src/algebra/homology/homology.lean

```diff
@@ -4,143 +4,192 @@ Released under Apache 2.0 license as described in the file LICENSE.
+  image_to_kernel_map C i ≫ kernel_map f (i + b) = image_map f i ≫ image_to_kernel_map C' i :=
 by { ext, simp }
 
 variables [has_cokernels V]
 
-/-- The `i`-th cohomology group of the cochain complex `C`. -/
-def cohomology (C : cochain_complex V) (i : ℤ) : V :=
-cokernel (image_to_kernel_map C (i-1))
+/-- The `i`-th homology group of the complex `C`. -/
+def homology_group (i : β) (C : homological_complex V b) : V :=
+cokernel (image_to_kernel_map C (i-b))
 
 variables [has_image_maps V]
 
-/-- A morphism of cochain complexes induces a morphism in cohomology at every degree. -/
-def cohomology_map {C C' : cochain_complex V} (f : C ⟶ C') (i : ℤ) :
-  C.cohomology i ⟶ C'.cohomology i :=
-cokernel.desc _ (kernel_map f (i - 1 + 1) ≫ cokernel.π _) $ by simp [induced_maps_commute_assoc]
+/-- A chain map induces a morphism in homology at every degree. -/
+def homology_map {C C' : homological_complex V b} (f : C ⟶ C') (i : β) :
+  C.homology_group i ⟶ C'.homology_group i :=
+cokernel.desc _ (kernel_map f (i - b + b) ≫ cokernel.π _) $
+  by simp [image_to_kernel_map_comp_kernel_map_assoc]
 
 @[simp, reassoc]
-lemma cohomology_map_condition {C C' : cochain_complex V} (f : C ⟶ C') (i : ℤ) :
-  cokernel.π (image_to_kernel_map C (i - 1)) ≫ cohomology_map f i =
-    kernel_map f (i - 1 + 1) ≫ cokernel.π _ :=
-by simp [cohomology_map]
+lemma homology_map_condition {C C' : homological_complex V b} (f : C ⟶ C') (i : β) :
+  cokernel.π (image_to_kernel_map C (i - b)) ≫ homology_map f i =
+    kernel_map f (i - b + b) ≫ cokernel.π _ :=
+by simp [homology_map]
 
 @[simp]
-lemma cohomology_map_id (C : cochain_complex V) (i : ℤ) :
-  cohomology_map (𝟙 C) i = 𝟙 (cohomology C i) :=
+lemma homology_map_id (C : homological_complex V b) (i : β) :
+  homology_map (𝟙 C) i = 𝟙 (C.homology_group i) :=
 begin
   ext,
-  simp only [cohomology_map_condition, kernel_map_id, category.id_comp],
+  simp only [homology_map_condition, kernel_map_id, category.id_comp],
   erw [category.comp_id]
 end
 
 @[simp]
-lemma cohomology_map_comp {C C' C'' : cochain_complex V} (f : C ⟶ C') (g : C' ⟶ C'') (i : ℤ) :
-  cohomology_map (f ≫ g) i = cohomology_map f i ≫ cohomology_map g i :=
+lemma homology_map_comp {C C' C'' : homological_complex V b} (f : C ⟶ C') (g : C' ⟶ C'') (i : β) :
+  homology_map (f ≫ g) i = homology_map f i ≫ homology_map g i :=
 by { ext, simp }
 
-/-- The cohomology functor from cochain complexes to `ℤ` graded objects in `V`. -/
-def cohomology_functor : cochain_complex V ⥤ graded_object ℤ V :=
-{ obj := λ C i, cohomology C i,
-  map := λ C C' f i, cohomology_map f i }
+variables (V)
+
+/-- The `i`-th homology functor from `β` graded complexes to `V`. -/
+@[simps]
+def homology (i : β) : homological_complex V b ⥤ V :=
+{ obj := λ C, C.homology_group i,
+  map := λ C C' f, homology_map f i, }
+
+/-- The homology functor from `β` graded complexes to `β` graded objects in `V`. -/
+@[simps]
+def homology_functor : homological_complex V b ⥤ graded_object β V :=
```

Comment:
One final question. What do you think of calling this `graded_homology`?
Otherwise, LGTM.

Reply:
I like it. Done.